lib/src/macos_window_utils_config.dart

```diff
@@ -0,0 +1,120 @@
+import 'package:flutter/widgets.dart';
+import 'package:macos_ui/macos_ui.dart';
+
+/// A class for configuring macOS window properties.
+///
+/// [toolbarStyle] is the style of the window toolbar. It should be
+/// [NSWindowToolbarStyle.expanded] if the app will have a title bar and
+/// [NSWindowToolbarStyle.unified] otherwise.
+///
+/// Example:
+/// ```dart
+/// final config = MacosWindowUtilsConfig(
+///   toolbarStyle: NSWindowToolbarStyle.expanded,
+/// );
+/// await config.apply();
+/// ```
+class MacosWindowUtilsConfig {
+  /// Creates a [MacosWindowUtilsConfig].
+  ///
+  /// The [toolbarStyle] is [NSWindowToolbarStyle.unified] by default. If the
+  /// app will have a title bar, use [NSWindowToolbarStyle.expanded] instead.
+  const MacosWindowUtilsConfig({
+    this.toolbarStyle = NSWindowToolbarStyle.unified,
+    this.doEnableFullSizeContentView = true,
+    this.doMakeTitlebarTransparent = true,
+    this.doHideTitle = true,
+    this.doRemoveMenubarInFullScreenMode = true,
+    this.doAutoHideToolbarAndMenuBarInFullScreenMode = true,
+  });
+
+  /// The style of the window toolbar.
+  ///
+  /// Defaults to [NSWindowToolbarStyle.unified]. Use
+  /// [NSWindowToolbarStyle.expanded] instead if the app will have a title bar.
+  final NSWindowToolbarStyle toolbarStyle;
+
+  /// Whether to enable the full-size content view.
+  final bool doEnableFullSizeContentView;
+
+  /// Whether to make the title bar transparent.
+  final bool doMakeTitlebarTransparent;
+
+  /// Whether to hide the title.
+  final bool doHideTitle;
+
+  /// Whether to remove the menubar in full-screen mode.
+  final bool doRemoveMenubarInFullScreenMode;
+
+  /// Whether to automatically hide the toolbar and menubar in full-screen mode.
+  final bool doAutoHideToolbarAndMenuBarInFullScreenMode;
```

Comment:
Can we rename these to drop the "do"? For example:
* `enableFullSizeContentView`
* `hideTitle`

Reply:
Done.